dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/deduplication/DeduplicationHelper.java

```diff
@@ -274,7 +283,8 @@ public String getUserAccessErrors(
       return "Missing data write access to one or more Relationship Types.";
     }
 
-    List<Enrollment> enrollments = enrollmentService.getEnrollments(mergeObject.getEnrollments());
```

Comment:
Can we use the existing `getEnrollments(EnrollmentOperationParams params)` instead of creating a new method?

Reply:
I thought about that, but for deduplication, we need access to all requested enrollments. If any are missing, we should raise an exception. The current method just skips any that aren't accessible, which isn't ideal. 
I could compare the requested enrollments to the final list to find any missing ones, but that means creating a new exception instead of using the ones from the access manager, which I'd rather stick with.
That being said, I'm not convinced by either solution.

Reply:
ok, lets just not forget about looking into this 😬